### PR TITLE
Merge paradox/scala/remoting-artery.md and java/remoting-artery.md

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #23052

## Rendered HTML:

- https://richard-akka-doc.firebaseapp.com/scala/remoting-artery.html
- https://richard-akka-doc.firebaseapp.com/java/remoting-artery.html

## I split this in two commits:

- **1st commit:** Identical contents in scala/remoting-artery.md and java/remoting-artery.md -> **this is to take the diff both in scala/remoting-artery.md and java/remoting-artery.md**

  - [scala/remoting-artery.md  (prev) vs. scala/remoting-artery.md (after)](https://github.com/richard-imaoka/akka/commit/de2953dc0d2a7a260c940d6e0733c72f593d4614#diff-6b5751e179e615bfced810beb83e3a83)
  - [java/remoting-artery.md (prev) vs. java/remoting-artery.md (after)](https://github.com/richard-imaoka/akka/commit/de2953dc0d2a7a260c940d6e0733c72f593d4614#diff-ea0d7e75a212482d96bea4003425ad3d)

- **2nd commit:** Replace java/remoting-artery.md with a symlink to scala/remoting-artery.md

### CODE REVIEW

File: akka-docs/src/main/paradox/java/remoting-artery.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ acts as a "server" to which arbitrary systems on the same network can connect to
 ## What is new in Artery
 
 Artery is a reimplementation of the old remoting module aimed at improving performance and stability. It is mostly
-source compatible with the old implementation and it is a drop-in replacement in many cases. Main features
+backwards compatible with the old implementation and it is a drop-in replacement in many cases. Main features
```
Comment (raboof) [inline akka-docs/src/main/paradox/java/remoting-artery.md]: Hmm 'source compatible' might be slightly more accurate here?

File: akka-docs/src/main/paradox/java/remoting-artery.md [hunk 2]
```diff
@@ -45,15 +45,23 @@ are also different.
...
+    </dependency>
+    ```
+    @@@
+
+Java
+:   @@@vars
+    ```
+    "com.typesafe.akka" %% "akka-remote" % "$akka.version$"
+    ```
+    @@@
```
Comment (raboof) [inline akka-docs/src/main/paradox/java/remoting-artery.md]: The Scala and Java snippet bodies are switched here

File: akka-docs/src/main/paradox/java/remoting-artery.md [hunk 3]
```diff
@@ -141,10 +149,21 @@ In the next sections the two alternatives are described in detail.
 
 `actorSelection(path)` will obtain an `ActorSelection` to an Actor on a remote node, e.g.:
 
-```
-ActorSelection selection =
-  context.actorSelection("akka://actorSystemName@10.0.0.1:25520/user/actorName");
-```
+Scala
+:   @@@vars
```
Comment (raboof) [inline akka-docs/src/main/paradox/java/remoting-artery.md]: Do we need `@@@vars` here?

File: akka-docs/src/main/paradox/java/remoting-artery.md [hunk 4]
```diff
@@ -284,7 +328,7 @@ so if network security is not considered as enough protection the classic remoti
 
 Best practice is that Akka remoting nodes should only be accessible from the adjacent network.
 
-It is also security best-practice to [disable the Java serializer](#disable-java-serializer-java-artery) because of
+It is also security best practice to @ref:[disable the Java serializer](#disabling-the-java-serializer) because of
```
Comment (raboof) [inline akka-docs/src/main/paradox/java/remoting-artery.md]: I think we prefer `@ref` without `:`

File: akka-docs/src/main/paradox/java/remoting-artery.md [hunk 5]
```diff
@@ -311,7 +355,7 @@ as a marker trait to user-defined messages.
 
 Untrusted mode does not give full protection against attacks by itself.
 It makes it slightly harder to perform malicious or unintended actions but
-it should be complemented with [disabled Java serializer](#disable-java-serializer-java-artery).
+it should be complemented with @ref[disabled Java serializer](#disabling-the-java-serializer)
```
Comment (raboof) [inline akka-docs/src/main/paradox/java/remoting-artery.md]: :+1:

File: akka-docs/src/main/paradox/java/remoting-artery.md [hunk 6]
```diff
@@ -321,9 +365,19 @@ Messages sent with actor selection are by default discarded in untrusted mode, b
...
+:   @@@vars
+    ```
+    akka.remote.artery.trusted-selection-paths = ["/user/receptionist", "/user/namingService"]
+    ```
+    @@@
+
+Java
+:   @@@vars
+    ```
+    akka.remote.artery..trusted-selection-paths = ["/user/receptionist", "/user/namingService"]
```
Comment (raboof) [inline akka-docs/src/main/paradox/java/remoting-artery.md]: A `.` too many here it seems

**Review by raboof — CHANGES_REQUESTED** [review]
Thanks for another nice merge! Some nitpicks here and there

File: akka-docs/src/main/paradox/java/remoting-artery.md (hunk 3 above)
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/java/remoting-artery.md]: Is it not only for this place but generally for in-line code snippets **_not_** using `${property}` values? 

If we should remove unnecessary `@@@vars`  from in-line snippets, I can create a separate PR updating **_other_** .md files. (Of course I can fix remoting-artery.md in this PR)

**Comment by richard-imaoka** [comment]
Sorry it seems like I left many of small issues to be fixed. I reflected your review comments - let me know if it looks better now.

File: akka-docs/src/main/paradox/scala/remoting-artery.md [hunk 7]
```diff
@@ -137,10 +150,18 @@ In the next sections the two alternatives are described in detail.
...
+:   ```
+    ActorSelection selection =
+      context.actorSelection("akka://actorSystemName@10.0.0.1:25520/user/actorName");
+    ```
+    
+Java
+:   ```
+    val selection =
+      context.actorSelection("akka://actorSystemName@10.0.0.1:25520/user/actorName")
+    ```
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/remoting-artery.md]: Scala/java snippets seem to be reversed here

File: akka-docs/src/main/paradox/scala/remoting-artery.md [hunk 8]
```diff
@@ -156,9 +177,19 @@ Unlike with earlier remoting, the protocol field is always *akka* as pluggable t
...
+    selection.tell("Pretty awesome feature", getSelf());
+    ```
+    @@@
+
+Java
+:   @@@vars
+    ```
+    selection ! "Pretty awesome feature"
+    ```
+    @@@
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/remoting-artery.md]: and here too

File: akka-docs/src/main/paradox/scala/remoting-artery.md [hunk 9]
```diff
@@ -317,9 +363,19 @@ Messages sent with actor selection are by default discarded in untrusted mode, b
...
+    akka.remote.artery.trusted-selection-paths = ["/user/receptionist", "/user/namingService"]
+    ```
+    @@@
+
+Java
+:   @@@vars
+    ```
+    akka.remote.artery.trusted-selection-paths = ["/user/receptionist", "/user/namingService"]
+    ```
+    @@@
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/remoting-artery.md]: Looks like this is the same for Java and Scala - perhaps make it just a normal snippet without tabs?

**Review by raboof — CHANGES_REQUESTED** [review]
Almost there 2 small tweaks needed!

**Review by raboof — APPROVED** [review]

**Comment by raboof** [comment]
Thanks, looks good!

**Pull request merged**